packages/amplication-git-utils/src/git/git.service.ts

```diff
@@ -1,207 +1,73 @@
-import { Injectable } from "@nestjs/common";
-
 import {
-  EnumGitOrganizationType,
-  EnumGitProvider,
-  EnumPullRequestMode,
   GithubFile,
-  GitResourceMeta,
-  PullRequestModule,
+  GitProviderArgs,
   RemoteGitOrganization,
   RemoteGitRepos,
   RemoteGitRepository,
+  File,
+  GetRepositoryArgs,
+  GetRepositoriesArgs,
+  CreateRepositoryArgs,
+  CreatePullRequestArgs,
+  GitProvider,
 } from "../types";
-import { AmplicationIgnoreManger } from "../utils/amplication-ignore-manger";
-import { Changes } from "octokit-plugin-create-pull-request/dist-types/types";
-import { join } from "path";
-import { AMPLICATION_IGNORED_FOLDER } from "./git.constants";
-import { GitServiceFactory } from "./git-service-factory";
+import { prepareFilesForPullRequest } from "../utils/prepare-files-for-pull-request";
+import { GitFactory } from "./git-factory";
 
-@Injectable()
-export class GitService {
-  constructor(private readonly gitServiceFactory: GitServiceFactory) {}
+export class GitClientService {
+  private provider: GitProvider;
 
-  async getReposOfOrganization(
-    gitProvider: EnumGitProvider,
-    installationId: string,
-    limit: number,
-    page: number
-  ): Promise<RemoteGitRepos> {
-    const gitService = this.gitServiceFactory.getService(gitProvider);
-    return await gitService.getOrganizationRepos(installationId, limit, page);
+  static createProvider(gitProviderArgs: GitProviderArgs): GitClientService {
+    const serviceInstance = new this();
+    serviceInstance.provider = GitFactory.getProvider(gitProviderArgs);
+    return serviceInstance;
   }
```

Comment:
We are using a class here, but we created a static function with the responsibility to create the class instance.
This is actually a `constructor` https://www.typescriptlang.org/docs/handbook/2/classes.html#constructors